R/derive_joined.R

```diff
@@ -423,4 +424,10 @@ derive_vars_joined <- function(dataset,
       )
     ) %>%
     remove_tmp_vars()
+
+  if (is.null(original_new_vars)) {
+    data_final <- data_final %>%
```

Comment:
@bundfussr this one addresses the NULL new_vars issue, don't think it's the most "elegant" but should work...?

Reply:
I think it is not in line with the documentation and intended behavior. If `new_vars = NULL` is specified, all variables from `dataset_add` should be added. However, in the example from the issue no variable is added. Of course, we can not add all variables because they are already in the input dataset. Thus an error should be issued.

Reply:
So in the case of all variables already in the input dataset, the desired result would be an error and no-return object at all?

Reply:
Yes, or more precise, if _any_ of the variables specified for `new_vars` is already in the input dataset, the function should stop with an error.

Reply:
Please see updated `if (is.null(new_vars))` at line 353 that implements suggestion and updated test with the corresponding messaging